marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/artifacts/core.mjs

```diff
@@ -26,7 +27,8 @@ const cachedArtifacts = {};
 const registeredArtifactTypes = {
     loadData: LoadData,
```

Comment:
What's the difference between a "loadData" artifact and a "loadDataSettings" artifact? I'm reading DHFPROD-4096, and the purpose of the new endpoints is to support only modifying the "settings" of a "loadData" config. That doesn't seem like it would mean that "settings" is a separate artifact type though, right?

Reply:
I understand, Conceptionally, the settings is not artififact type. One loadData artifact can be associated with only one setting. I just wanted to separate different implementation details because of different validation (different required fields, document properties) and query.  I can certainly add different methods for settings in the loadData.mjs instead of have a new type implmetation (loadDataSetting.mjs).

Reply:
If the settings is not an artifact type, then this will permanently create confusion by including it in a variable called "registeredArtifactTypes". 

Is the goal for the web client to have an endpoint where it only needs to pass in a portion of a LoadData object? If so, couldn't the middle tier wrap it in an otherwise-empty LoadData object? And the LoadData module would be able to figure out that only the settings are being passed in. That seems much easier to understand vs corrupting the meaning of an "artifact" by including both "LoadData" and "LoadDataSettings" in it.

Reply:
@rjrudin i removed the confusion in the latest check-in...